### PR TITLE
migration: use ERROR_INT()

### PR DESCRIPTION
**Pull request by jlevon**

### CODE REVIEW

File: lib/migration.c [hunk 1]
```diff
@@ -289,7 +289,7 @@ handle_data_offset(vfu_ctx_t *vfu_ctx, struct migration *migr,
         if (is_write) {
             /* TODO writing to read-only registers should be simply ignored */
             vfu_log(vfu_ctx, LOG_ERR, "bad write to migration data_offset");
-            return -EINVAL;
+            return ERROR_INT(EINVAL);
         }
         ret = migr->callbacks.prepare_data(vfu_ctx, offset, NULL);
```
Comment (tmakatos) [inline lib/migration.c:294]: This is a good opportunity to update `prepare_data` documentation specifing that -1 must be returned on failure and errno must be set.

File: lib/migration.c [hunk 2]
```diff
@@ -442,14 +442,14 @@ migration_region_access(vfu_ctx_t *vfu_ctx, char *buf, size_t count,
             vfu_log(vfu_ctx, LOG_WARNING,
                     "bad access to dead space %#lx-%#lx in migration region",
                     pos, pos + count - 1);
-            return -EINVAL;
+            return ERROR_INT(EINVAL);
         }
 
         pos -= migr->data_offset;
         if (is_write) {
             ret = migr->callbacks.write_data(vfu_ctx, buf, count, pos);
```
Comment (tmakatos) [inline lib/migration.c:450]: Check that errno is non-zero?

File: lib/migration.c [hunk 3]
```diff
@@ -461,16 +461,13 @@ migration_region_access(vfu_ctx_t *vfu_ctx, char *buf, size_t count,
              * Does this mean that partial reads are not allowed?
              */
             ret = migr->callbacks.read_data(vfu_ctx, buf, count, pos);
-            if (ret == -1) {
-                ret = -errno;
+            if (ret < 0) {
```
Comment (tmakatos) [inline lib/migration.c:464]: ditto

**Review by tmakatos — APPROVED** [review]
Looks good, minor comments and suggestions.

File: lib/migration.c (hunk 1 above)
Comment (jlevon) [reply lib/migration.c:294]: yep thanks

File: lib/migration.c (hunk 2 above)
Comment (jlevon) [reply lib/migration.c:450]: as per the other set, I don't think it really makes sense to try and check this.

**Review by tmakatos — APPROVED** [review]

**Review by tmakatos — APPROVED** [review]

**Pull request merged**